controllers/storagecluster/external_resources.go

```diff
@@ -254,6 +254,8 @@ func (obj *ocsExternalResources) ensureCreated(r *StorageClusterReconciler, inst
 			return r.onboardConsumer(instance, externalClusterClient)
 		} else if instance.Status.Phase == statusutil.PhaseOnboarding {
 			return r.acknowledgeOnboarding(instance, externalClusterClient)
+		} else if instance.Status.Phase == statusutil.PhaseReady {
+			return r.enableReportStatusCronJob(instance, externalClusterClient)
```

Comment:
Why do we need to run this only when we are Ready? why can't we just call this in the else section?
Also, this change means that the code below (the next else if)will not apply to Ready state anymore - which I don't think was your original intention.

Reply:
> Why do we need to run this only when we are Ready?
Because once the external storageCluster is created that means there is a corresponding storageConsumer on provider so we can hit the gRPC endpoint to update lastHeartbeat timestamp. If storageCluster is in onboarding or connecting state then there is no guarantee that storageConsumer resource is created. So I think it is better to run the cronJob after storageCluster is in ready state.

> why can't we just call this in the else section?
There are other phases as well such as error, ignored and not ready. If storageCluster is in this phase just after creation then I don't think we should start the job as corresponding storageConsumer is not created.